baybe/acquisition/base.py

```diff
@@ -69,18 +70,24 @@ def _add_deprecation_hook(hook):
     """
 
     def added_deprecation_hook(val: Union[dict, str], cls: type):
-        UCB_DEPRECATIONS = {
-            "VarUCB": "UpperConfidenceBound",
-            "qVarUCB": "qUpperConfidenceBound",
-        }
-        if (entry := val if isinstance(val, str) else val["type"]) in UCB_DEPRECATIONS:
-            warnings.warn(
-                f"The use of `{entry}` is deprecated and will be disabled in a "
-                f"future version. The get the same outcome, use the new "
-                f"`{UCB_DEPRECATIONS[entry]}` class instead with a beta of 100.0.",
-                DeprecationWarning,
-            )
-            val = {"type": UCB_DEPRECATIONS[entry], "beta": 100.0}
+        # Backwards-compatibility needs to be ensured only for deserialization from
+        # base class using string-based type specifiers as listed below,
+        # since the concrete classes were available only after the change.
+        if is_abstract(cls):
```

Comment:
Can you quickly elaborate what exactly you mean (sorry, confused). Just as an explanation: was simply the easiest way for me to get all deserialization tests working, and because it's a deprecation mechanism that will vanish soon, I didn't care too much if there is a more elegant version. At least without it, you run into problems. Do you have a concrete suggestion how you would like to have it instead?

Reply:
as far as i can see this indent limits the scope of the function, which does not seem necessary to me, os is it?

Reply:
It is. Try do delete it and run `test_deserialization`, which will fail and again underpins my point that these tests are indeed needed and not yet fully covered by our other test cases. But I'm more than happy to change to a better solution if you propose one. As I said, was mostly because it was just the simplest way to get it running and we are only talking about deprecation code. Let me know how else you would prefer to make the test succeed